firestore/integration_test_internal/src/settings_test.cc

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 Google LLC
+
+#include <future>
+#include <map>
+#include <string>
+
+#include "firebase/firestore.h"
+#include "firestore_integration_test.h"
+
+namespace firebase {
+namespace firestore {
+
+using SettingsTest = FirestoreIntegrationTest;
+
+TEST_F(SettingsTest, Equality) {
+  int64_t five_mb = 5 * 1024 * 1024;
+  int64_t six_mb = 6 * 1024 * 1024;
+
+  Settings settings1;
+  settings1.set_host("foo");
+  settings1.set_ssl_enabled(true);
+  settings1.set_persistence_enabled(true);
+  settings1.set_cache_size_bytes(five_mb);
+
+  Settings settings2;
+  settings2.set_host("bar");
+  settings2.set_ssl_enabled(true);
+  settings2.set_persistence_enabled(true);
+  settings2.set_cache_size_bytes(five_mb);
+
+  Settings settings3;
+  settings3.set_host("foo");
+  settings3.set_ssl_enabled(false);
+  settings3.set_persistence_enabled(true);
+  settings3.set_cache_size_bytes(five_mb);
+
+  Settings settings4;
+  settings4.set_host("foo");
+  settings4.set_ssl_enabled(true);
+  settings4.set_persistence_enabled(false);
+  settings4.set_cache_size_bytes(five_mb);
+
+  Settings settings5;
+  settings5.set_host("foo");
+  settings5.set_ssl_enabled(true);
+  settings5.set_persistence_enabled(true);
+  settings5.set_cache_size_bytes(six_mb);
```

Comment:
Can you also add an object that is equal to one of the objects above? Currently, the only test for equality that returns `true` is when an object is compared to itself, ideally we'd also compare two different objects.

Reply:
Done